wallet-new/src/Cardano/Wallet/Kernel/DB/Sqlite.hs

```diff
@@ -447,6 +448,34 @@ clearMetaDB conn = do
 putTxMeta :: Sqlite.Connection -> Kernel.TxMeta -> IO ()
 putTxMeta conn txMeta = void $ putTxMetaT conn txMeta
 
+-- | Clear some metadata from the database
+deleteTxMetas
+    :: Sqlite.Connection
+        -- | Database Handle
+    -> Core.Address
+        -- | Target wallet
+    -> Maybe Word32
+        -- |  A target account index. If none, delete metas for all accounts
+    -> IO ()
+deleteTxMetas conn walletId mAccountIx = do
+    runBeamSqlite conn $ SQL.runDelete $ SQL.delete (_mDbMeta metaDB) $ \meta ->
```

Comment:
We need also to manually delete from input and output tables. Ideally we would use Foreign Keys to let sqlite cascade deletes automagically, but unfortunately beam does not support them. A solution is to follow this:
- get all txids from this wallet/account.
- delete all inputs/outputs of these txids.
- delete metas as you have done.

All above should be wrapped in a `withTransaction`, to make sure we roll back in case of errors.

Reply:
hmm actually this may be more tricky. We want to clean inputs and outputs only if the deleted wallet/account is the last one with this txid.

Reply:
Right. There's something here indeed. Nees to figure out where do we exactly rely on those inputs and outputs and see.